### ISSUE TITLE
Create ruleset.xml based on current state of the code

### ISSUE THREAD

**Issue by lsolesen** [squizlabs/PHP_CodeSniffer#987]
I've been contributing phpcs.xml files with rulesets to a couple of open source libraries now. I do the following:

```
phpcs --standard=phpcs.xml src -s
```

Then I add the individual sniffs to `phpcs.xml`

``` xml
<?xml version="1.0"?>
<!--suppress XmlUnboundNsPrefix -->
<ruleset name="PHP-SDK">
 <description>Coding standard ruleset based on the PSR-2 coding standard.</description>
 <rule ref="PSR2"/>
 <rule ref="Generic.Files.LineLength.TooLong">
  <severity>0</severity>
 </rule>
```

Until the phpcs test passes. When it passes I start to remove rules one by one to fix the issues.

However, it can be tedious to create the file. How about adding a tool like:

```
phpcs-ruleset --standard=PSR2 --ruleset-name=phpcs.xml src
```

Then the ruleset based on the current code is automatically created.

**Comment by gsherwood** [comment]
I'm going to close this because I don't think I could take on another project like this.

But I'd suggest using the info report if you want to examine a codebase and determine how code is written. There is an example (and a tip for how to check the most code blocks) here: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Reporting#printing-an-information-report

I use this report to inspect code from a number of PHP projects, and determine a common PHP style, over here: https://squizlabs.github.io/PHP_CodeSniffer/analysis/
